src/main/java/org/maxgamer/quickshop/eventmanager/QSEventManager.java

```diff
@@ -0,0 +1,177 @@
+    public void serverReloaded(ServerLoadEvent event) {
+        this.rescan();
+    }
+
+    @EventHandler(priority = EventPriority.MONITOR, ignoreCancelled = true)
+    public void pluginDisable(PluginDisableEvent event) {
+        this.rescan();
+    }
+
+    @EventHandler(priority = EventPriority.MONITOR, ignoreCancelled = true)
+    public void pluginEnable(PluginEnableEvent event) {
+        this.rescan();
+    }
+
+    private synchronized void rescan() {
+        this.ignoredListener.clear();
+        plugin
+                .getConfig()
+                .getStringList("shop.protection-checking-listener-blacklist")
+                .forEach(
+                        input -> {
+                            if (StringUtils.isEmpty(input)) {
+                                return;
+                            }
+                            try {
+                                Class<?> clazz = Class.forName(input);
+                                this.ignoredListener.add(new ListenerContainer(clazz, input));
+                                Util.debugLog("Successfully added blacklist: [CLAZZ] " + clazz.getName());
+                            } catch (ClassNotFoundException ignored) {
+                                this.ignoredListener.add(new ListenerContainer(null, input));
+                                Util.debugLog("Successfully added blacklist: [DYNAMIC] " + input);
+                            }
+                        });
+    }
+
+    public void callEvent(Event event) {
+        if (event.isAsynchronous()) {
+            if (Thread.holdsLock(Bukkit.getPluginManager())) {
+                throw new IllegalStateException(
+                        event.getEventName()
+                                + " cannot be triggered asynchronously from inside synchronized code.");
+            }
+            if (Bukkit.getServer().isPrimaryThread()) {
+                throw new IllegalStateException(
+                        event.getEventName()
+                                + " cannot be triggered asynchronously from primary server thread.");
+            }
+        } else {
+            if (!Bukkit.getServer().isPrimaryThread()) {
+                throw new IllegalStateException(
+                        event.getEventName() + " cannot be triggered asynchronously from another thread.");
+            }
+        }
+
+        fireEvent(event);
+    }
+
+    private void fireEvent(Event event) {
+        HandlerList handlers = event.getHandlers();
+        RegisteredListener[] listeners = handlers.getRegisteredListeners();
+
+        for (RegisteredListener registration : listeners) {
+            if (!registration.getPlugin().isEnabled()) {
+                continue;
+            }
+            Class<?> regClass = registration.getListener().getClass();
+            if (this.ignoredListener.stream().anyMatch(listenerContainer -> listenerContainer.matches(regClass, registration.getPlugin()))) {
```

Comment:
Replace with legacy way?  since is a high usages method.

Reply:
but drop the stream will no-longer use lambda
it will break continue logic

but you're right

Reply:
done